lib/pantry/helpers/dairy_category_helper.rb

```diff
@@ -0,0 +1,171 @@
+require 'set'
+
+module Pantry
+
+  module Helpers
+
+    class DairyCategoryHelper
+      class << self
+
+        def categorize(item_name)
+          categories_max = 0
+          category_name = ''
+          categories.each do |k,v|
+            set = method(k).to_proc.call
+
+            count = 0
+            item_name.split(' ').each do |word|
+              if set.include? word
+                count += 1
+              end
+            end
+
+            if count > categories_max
+              category_name = k
+            end
+          end
+          category_name.empty? ? :other : category_name
+        end
+
+        def time_til_warning(category:)
+          categories[category][:warning]
+        end
+
+        def time_til_expiration(category:)
+          categories[category][:expiring]
+        end
+
+        def categories
+          categories = {
+            milk: {
+              warning: 5,
+              expiring: 7
+            },
+            cream: {
+              warning: 7,
+              expiring: 14
+            },
+            butter: {
+              warning: 10,
+              expiring: 14
+            },
+            margarine: {
+              warning: 25,
+              expiring: 30
+            },
+            cheese: {
+              warning: 25,
+              expiring: 30
+            },
+            yogurt: {
+              warning: 25,
+              expiring: 30
+            },
+            other: {
+              warning: 0,
+              expiring: 0
+            }
+          }
+        end
```

Comment:
I'd move this hash into a constant so it's built and saved once, rather than every time this method is called.